Project - Rachel Jordan 

In [3]:
getwd()

[1] "/Users/rachelcjordan/notebook"

In [2]:
setwd("~/notebook")

install.packages("igraph")

In [6]:
library("igraph")

In [4]:
load("/Users/rachelcjordan/Downloads/astrocollab.Rdata")

In [5]:
summary(astrocollab)

IGRAPH UNW- 16706 121251 -- Astrophysics collaborations
+ attr: name (g/c), Author (g/c), Citation (g/c), URL (g/c),
| Description (g/c), name (v/c), weight (e/n)


This is a undirected, named, and weighted graph this has 16706 vertices and 121251 edges 
attributes of the graph include: name, author, citation, URL, description, name, weight



In [12]:
V(astrocollab) #captures the vertices in order 
#16706/16706 vertices

This graph was created by an old(er) igraph version.
  Call upgrade_graph() on it to use with the current igraph version
  For now we convert it on the fly...
This graph was created by an old(er) igraph version.
  Call upgrade_graph() on it to use with the current igraph version
  For now we convert it on the fly...


+ 16706/16706 vertices, named:
    [1] BIERMANN, PL                     STANEV, TKGT                    
    [3] GOLDMAN, I                       WANDEL, A                       
    [5] PILDIS, RA                       BREGMAN, JN                     
    [7] EVRARD, AE                       SWARTZ, DA                      
    [9] SUTHERLAND, PG                   HARKNESS, RP                    
   [11] BECKER, W                        BARZIER, KTS                    
   [13] TRUEMPER, J                      ANGELANTONJ, C                  
   [15] AMENDOLA, L                      LITTERIO, M                     
   [17] OCCHIONERO, F                    LOEB, A                         
   [19] BRAINERD, JJ                     MEEGAN, CA                      
+ ... omitted several vertices

In [13]:
upgrade_graph(astrocollab)

IGRAPH UNW- 16706 121251 -- Astrophysics collaborations
+ attr: name (g/c), Author (g/c), Citation (g/c), URL (g/c),
| Description (g/c), name (v/c), weight (e/n)
+ edges (vertex names):
 [1] BIERMANN, PL  --STANEV, TKGT   GOLDMAN, I    --WANDEL, A     
 [3] PILDIS, RA    --BREGMAN, JN    PILDIS, RA    --EVRARD, AE    
 [5] BREGMAN, JN   --EVRARD, AE     SWARTZ, DA    --SUTHERLAND, PG
 [7] SWARTZ, DA    --HARKNESS, RP   SUTHERLAND, PG--HARKNESS, RP  
 [9] BECKER, W     --BARZIER, KTS   BECKER, W     --TRUEMPER, J   
[11] BARZIER, KTS  --TRUEMPER, J    ANGELANTONJ, C--AMENDOLA, L   
[13] ANGELANTONJ, C--LITTERIO, M    AMENDOLA, L   --LITTERIO, M   
+ ... omitted several edges

In [14]:
myastro = upgrade_graph(astrocollab)

In [15]:
myastro

IGRAPH UNW- 16706 121251 -- Astrophysics collaborations
+ attr: name (g/c), Author (g/c), Citation (g/c), URL (g/c),
| Description (g/c), name (v/c), weight (e/n)
+ edges (vertex names):
 [1] BIERMANN, PL  --STANEV, TKGT   GOLDMAN, I    --WANDEL, A     
 [3] PILDIS, RA    --BREGMAN, JN    PILDIS, RA    --EVRARD, AE    
 [5] BREGMAN, JN   --EVRARD, AE     SWARTZ, DA    --SUTHERLAND, PG
 [7] SWARTZ, DA    --HARKNESS, RP   SUTHERLAND, PG--HARKNESS, RP  
 [9] BECKER, W     --BARZIER, KTS   BECKER, W     --TRUEMPER, J   
[11] BARZIER, KTS  --TRUEMPER, J    ANGELANTONJ, C--AMENDOLA, L   
[13] ANGELANTONJ, C--LITTERIO, M    AMENDOLA, L   --LITTERIO, M   
+ ... omitted several edges

Could run this: but is entirely too big
vertex_attr(myastro)

In [18]:
adjmatrix = as_adjacency_matrix(myastro) #adjacency matrix

In [19]:
adjmatrix

   [[ suppressing 32 column names ‘BIERMANN, PL’, ‘STANEV, TKGT’, ‘GOLDMAN, I’ ... ]]
   [[ suppressing 32 column names ‘BIERMANN, PL’, ‘STANEV, TKGT’, ‘GOLDMAN, I’ ... ]]
   [[ suppressing 32 column names ‘BIERMANN, PL’, ‘STANEV, TKGT’, ‘GOLDMAN, I’ ... ]]
   [[ suppressing 32 column names ‘BIERMANN, PL’, ‘STANEV, TKGT’, ‘GOLDMAN, I’ ... ]]
   [[ suppressing 32 column names ‘BIERMANN, PL’, ‘STANEV, TKGT’, ‘GOLDMAN, I’ ... ]]
   [[ suppressing 32 column names ‘BIERMANN, PL’, ‘STANEV, TKGT’, ‘GOLDMAN, I’ ... ]]


16706 x 16706 sparse Matrix of class "dgCMatrix"
                                                                                        
BIERMANN, PL             . 1 . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
STANEV, TKGT             1 . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
GOLDMAN, I               . . . 1 . . . . . . . . . . . . . . . . . . . . . . . . . . . .
WANDEL, A                . . 1 . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
PILDIS, RA               . . . . . 1 1 . . . . . . . . . . . . . . . . . . . . . . . . .
BREGMAN, JN              . . . . 1 . 1 . . . . . . . . . . . . . . . . . . . . . . . . .
EVRARD, AE               . . . . 1 1 . . . . . . . . . . . . . . . . . 1 1 . . . . . . .
SWARTZ, DA               . . . . . . . . 1 1 . . . . . . . . . . . . . . . . . . . . . .
SUTHERLAND, PG           . . . . . . . 1 . 1 . . . . . . . . . . . . . . . . . . . . . .
HARKNESS, RP             . . . . . . . 1 1 . . . . . . . . . 

In [1]:
plot(myastro)

ERROR: Error in plot(myastro): object 'myastro' not found


In [ ]:
degree(myastro)

Above, this gives us the number of degrees which will the sum of in and out degrees for each node. 